packages/modules/viewer-react/src/services/iModel/ComponentConnection.ts

```diff
@@ -0,0 +1,260 @@
+ *
+ * @See https://docs.bentley.com/LiveContent/web/AECOsim%20Building%20Designer%20Help-v2/en/GUID-CCDE2FF6-C63E-4876-B3AE-5F2B33EF8547.html
+ * for more about components.
+ */
+
+export class ComponentConnection extends IModelConnection {
+  public iModelProps: IModelConnectionProps;
+  private _documentId: string;
+  private _isClosed?: boolean = false;
+
+  public override get isClosed(): boolean {
+    return this._isClosed ? true : false;
+  }
+
+  constructor(iModelProps: IModelConnectionProps) {
+    super(iModelProps);
+
+    this.iModelProps = iModelProps;
+
+    if (!iModelProps.changeset?.id) {
+      throw new IModelError(
+        BentleyStatus.ERROR,
+        "DocumentId is required for a ComponentConnection"
+      );
+    }
+    this._documentId = iModelProps.changeset?.id;
+  }
+
+  /** Type guard for instanceof `ComponentConnection */
+  public isComponentConnection(): this is ComponentConnection {
+    return true;
+  }
+
+  public static async openRemote(
+    contextId: string,
+    componentId: string,
+    documentId: string
+  ): Promise<ComponentConnection> {
+    const iModelProps = {
+      iTwinId: contextId,
+      iModelId: componentId,
+      changeset: { id: documentId },
+    };
+
+    const routingContext =
+      IModelRoutingContext.current || IModelRoutingContext.default;
+    const props = await ComponentConnection.callOpen(
+      iModelProps,
+      routingContext
+    );
+    const connection = new ComponentConnection(props);
+
+    RpcManager.setIModel(connection);
+    connection.routingContext = routingContext;
+    RpcRequest.notFoundHandlers.addListener(
+      connection._reopenConnectionHandler
+    );
+
+    IModelConnection.onOpen.raiseEvent(connection);
+    return connection;
+  }
+
+  private static async callOpen(
+    iModelProps: ComponentProps,
+    routingContext: IModelRoutingContext
+  ): Promise<IModelConnectionProps> {
+    // Try opening the iModel repeatedly accommodating any pending responses from the backend.
+    // Waits for an increasing amount of time (but within a range) before checking on the pending request again.
+    const connectionRetryIntervalRange = { min: 100, max: 5000 }; // in milliseconds
+    let connectionRetryInterval = Math.min(
+      connectionRetryIntervalRange.min,
+      IModelConnection.connectionTimeout
+    );
+
+    const openForReadOperation = RpcOperation.lookup(
+      IModelReadRpcInterface,
+      "getConnectionProps"
+    );
+    if (!openForReadOperation) {
+      throw new IModelError(
+        BentleyStatus.ERROR,
+        "IModelReadRpcInterface.getConnectionProps() is not available"
+      );
+    }
+    openForReadOperation.policy.retryInterval = () => connectionRetryInterval;
+
+    Logger.logTrace(loggerCategory, `IModelConnection.open`, iModelProps);
+    const startTime = Date.now();
+
+    const removeListener = RpcRequest.events.addListener(
+      // eslint-disable-next-line deprecation/deprecation
```

Comment:
The explicit types for type and request probably aren't needed, which also makes this eslint-disable comment unneeded.  The following eslint-disable comment is obviously still needed due to using `RpcRequestEvent`.